modules/gdprEnforcement.js

```diff
@@ -191,13 +192,21 @@ export function validateRules(rule, consentData, currentModule, gvlId) {
   }
   const vendorConsentRequred = rule.enforceVendor && !((gvlId === VENDORLESS_GVLID || (rule.softVendorExceptions || []).includes(currentModule)));
   const {purpose, vendor} = getConsent(consentData, ruleOptions.type, ruleOptions.id, gvlId);
-  return (!rule.enforcePurpose || purpose) && (!vendorConsentRequred || vendor);
+
+  let validation = (!rule.enforcePurpose || purpose) && (!vendorConsentRequred || vendor);
+
+  if (gvlId === FIRST_PARTY_GVLID) {
+    validation = (!rule.enforcePurpose || !!deepAccess(consentData, `vendorData.publisher.consents.${ruleOptions.id}`));
```

Comment:
One question..  when FIRST_PARTY_GVLID is present I’m guessing we don’t need to worry about vendor for this validation? (looks to be the case with VENDORLESS_GVLID as well, so I was following down that road)

Reply:
yep, in this case purposeConsent is required, vendorConsent is not, and the flag to check is `publisher.consents`